Convert two react class components to functional components

drob opened the pull request:
Refactors two react classes to use functional components instead of class components, per #1423.

## Changes

- Refactors `StringTrimmedControl` to use functional components instead of class components.

## Testing

1. PR checks should pass.
2. Any regressions should show up in the netlify preview.

## Notes

- I'm new to some of these libraries, so apologies if I'm doing something non-idiomatic. (Please let me know, if so!)
- There are three more class components in `admin/src/widgets`, which I'm planning on handling in a subsequent PR.

anselmbradford commented:
Thanks so much for the contribution @drob !

If I visit the deploy preview for this and attempt to edit a page, I'm getting 

<img width="1574" alt="Screen Shot 2022-10-26 at 10 06 55 AM" src="https://user-images.githubusercontent.com/704760/198048418-bc44c372-3719-4acc-8a61-268285cbfb00.png">

Do you see why this might be the case? Maybe a default value not set on the `useState` call?

P.S. if you have any interest in trying other conversions from class to functional components in a different context, https://github.com/cfpb/ccdb5-ui/ also needs this task done.

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
-    setActiveStyle: PropTypes.func.isRequired,
-    setInactiveStyle: PropTypes.func.isRequired,
-  };
+export default function StringTrimmedControl({
+  onChange,
+  forID,
+  value,
+  classNameWrapper,
+  setActiveStyle,
+  setInactiveStyle,
+}) {
```
this syntax should also be valid:

```
export default const StringTrimmedControl = ({
  onChange,
  forID,
  value = '',
  classNameWrapper,
  setActiveStyle,
  setInactiveStyle,
}) => {
```

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
+  setActiveStyle,
+  setInactiveStyle,
+}) {
+  const [state, setState] = React.useState({ value: value || '' });
```
Since this is an input, i would just make the value inside of useState always be a `string`

flacoman91 replied:
If you're writing a lot of code that will use any of these react /redux hooks, you can also add this to the imports at the top

import React, { useState } from 'react';

so you can just do

const [foo, setFoo] = useState('');

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
+      id={forID}
+      className={classNameWrapper}
+      value={state.value}
+      onChange={(event) => handleChange(event)}
```
this can just be 
`onChange={handleChange}`

like how you have it down in StringWithInstructions

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
+      type="text"
+      id={forID}
+      className={classNameWrapper}
+      value={state.value}
```
if you have the value changed from above from being an object, you don't need to make it state.value here.

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
+StringTrimmedControl.propTypes = {
+  onChange: PropTypes.func.isRequired,
+  forID: PropTypes.string,
+  value: PropTypes.node,
```
value should probably be a string here since that's going to be the default value that will go into the input.

drob commented:
@flacoman91 pushed a second commit that addresses your comments. Agreed these are all improvements. I was trying to keep the diff as minimal as possible – just class -> functional – and avoid doing additional refactoring while in there.

@anselmbradford I don't think I can reproduce that, since I'm not a repo contributor and so can't auth with the CMS. Thanks for the pointer to ccb5-ui, by the way! Let me know if you come across others.

flacoman91 commented on docs/admin/src/widgets/StringTrimmed.js:
```diff
@@ -1,39 +1,39 @@
-  state = {
-    value: this.props.value || '',
+  const handleChange = (event) => {
+    onChange(event.target.value.trim());
```
I think this onChange line should be deleted

and the next line can be:
setValue(event.target.value.trim())

drob replied:
Is that safe? Seems like it would change behavior vs the code that's there now, if an `onChange` function is passed in.

flacoman91 replied:
Sorry, you're right.

 I didn't see it was being passed in and was confused by the onChange inside of the input

flacoman91 commented:
> @flacoman91 pushed a second commit that addresses your comments. Agreed these are all improvements. I was trying to keep the diff as minimal as possible – just class -> functional – and avoid doing additional refactoring while in there.
> 
> @anselmbradford I don't think I can reproduce that, since I'm not a repo contributor and so can't auth with the CMS. Thanks for the pointer to ccb5-ui, by the way! Let me know if you come across others.

Hi @drob!

Thanks for your contributions!  This is great stuff.  No worries.
Any help and feedback on our code is greatly appreciated.

I'm one of the maintainers of the [Consumer Complaint Database](https://www.consumerfinance.gov/data-research/consumer-complaints/search/).  If you ever wanted to take a look of the ins and outs of how that app works, you're more than welcome to check it out.  We're in the process of upgrading it to functional components, rewriting unit tests using react-testing-library, and also upgrading it to React 18.  I can also create issues with small tasks that anyone can pick up.
